pkg/controller/kubemon/reconciler.go

```diff
@@ -0,0 +1,187 @@
+package kubemon
+
+import (
+	"context"
+	"time"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/apis/dynatrace/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/customproperties"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/dtpods"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/dtpullsecret"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/dtversion"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/pkg/dtclient"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	Name                   = "kubernetes-monitoring"
+	annotationTemplateHash = "internal.activegate.dynatrace.com/template-hash"
+)
+
+type Reconciler struct {
+	client.Client
+	scheme    *runtime.Scheme
+	dtc       dtclient.Client
+	log       logr.Logger
+	token     *corev1.Secret
+	instance  *v1alpha1.DynaKube
+	apiReader client.Reader
+}
+
+func NewReconciler(clt client.Client, apiReader client.Reader, scheme *runtime.Scheme, dtc dtclient.Client, log logr.Logger, token *corev1.Secret, instance *v1alpha1.DynaKube) *Reconciler {
+	return &Reconciler{
+		Client:    clt,
+		apiReader: apiReader,
+		scheme:    scheme,
+		dtc:       dtc,
+		log:       log,
+		token:     token,
+		instance:  instance,
+	}
+}
+
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
No need for the `request` argument. I presume `reconcile.Result` is actually being in use, so that can stay.

Reply:
removed `request`-Parameter from all `Reconcile`-functions but `activegate.Reconcile`